rx_signal_processing/dsp.cu

```diff
@@ -327,6 +330,26 @@ namespace {
         }
       };
 
+      // Add our beamformed IQ data to the processed data packet that gets sent to data_write.
+      for (uint32_t beam_count=0; beam_count<beam_direction_counts[i]; beam_count++) {
+        auto beam = dataset->add_beamformedsamples();
+        beam->set_beamnum(beam_count);
+
+        for (uint32_t sample=0; sample<num_samples_after_dropping; sample++){
+          auto main_sample = beam->add_mainsamples();
+          main_sample->set_real(beamformed_samples_main[beamformed_offset + sample].x);
+          main_sample->set_imag(beamformed_samples_main[beamformed_offset + sample].y);
+
+          if (dp->sig_options.get_interferometer_antenna_count() > 0) {
+            auto intf_sample = beam->add_intfsamples();
+            intf_sample->set_real(beamformed_samples_intf[beamformed_offset + sample].x);
+            intf_sample->set_imag(beamformed_samples_intf[beamformed_offset + sample].y);
+          }
+      }
+
+      }
+      beamformed_offset += beam_direction_counts[i];
```

Comment:
Should this be before the closing of the for(beam_count; beam_count< beam_direction_counts[i] ...) loop?

Reply:
No, we need to keep track of offsets as we move along frequencies, as different slices can have different # of beams.

Reply:
OK, your tab spacing is incorrect here which is throwing me off. There are  two closing braces that are at the same indentation level. Maybe since there are so many loops/scopes here you could add comments to the closing braces to indicate which scope they are closing.

Reply:
Okay I think I fixed this.